Add continuous integration example and move use cases to the appendix (#78)
oauth-wg/oauth-identity-chaining: draft-ietf-oauth-identity-chaining.md

--- back

# Use cases

> This sections outlines some use cases the authors have identified where identity chaining as described in this document should be used. This section is not complete and other use cases not mentioned here are also valid.

Reviewer: Suggest a minor change in wording here (that also anticipates the likely merge of #76):

```suggestion
This sections outlines some use cases where the identity and authorization chaining described in this document can be applied. This section is not complete and other use cases not mentioned here are also valid.
```

## Preserve User Context across Multi-cloud, Multi-Hybrid environments
A user attempts to access a service that is implemented as a number of on-premise and cloud-based microservices. Both the on-premise and cloud-based services are segmented by multiple trust boundaries that span one or more on-premise or cloud service environments. Every microservice can apply an authorization policy that takes the context of the original user, as well as intermediary microservices into account, irrespective of where the microservices are running and even when a microservice in one trust domain calls another service in another trust domain.

> ## Continuous integration accessing external resources

Reviewer: per Dr. Kelley W Burgin:

```suggestion
## Continuous Integration Accessing External Resources
```

> A continuous integration system needs to access external resources, for example to upload an artifact or to run tests. These resources are protected by different authorization servers. The identity information of the build, for example metadata such as commit hashes or repository, should be preserved and carried across the domain boundary. This not just prevents maintaining credentials it also allows fined grained access control at the resource.

Reviewer 2: replace "fined" with "fine"
